streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java

```diff
@@ -0,0 +1,519 @@
+            testMetricByName(listMetricStore, PUT_IF_ABSENT_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, GET_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, GET_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, DELETE_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, DELETE_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, PUT_ALL_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, PUT_ALL_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, ALL_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, ALL_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, RANGE_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, RANGE_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, FLUSH_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, FLUSH_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, RESTORE_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, RESTORE_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, PUT_RATE, 2);
+            testMetricByName(listMetricStore, PUT_TOTAL, 2);
+            testMetricByName(listMetricStore, PUT_IF_ABSENT_RATE, 2);
+            testMetricByName(listMetricStore, PUT_IF_ABSENT_TOTAL, 2);
+            testMetricByName(listMetricStore, GET_RATE, 2);
+            testMetricByName(listMetricStore, DELETE_RATE, 2);
+            testMetricByName(listMetricStore, DELETE_TOTAL, 2);
+            testMetricByName(listMetricStore, PUT_ALL_RATE, 2);
+            testMetricByName(listMetricStore, PUT_ALL_TOTAL, 2);
+            testMetricByName(listMetricStore, ALL_RATE, 2);
+            testMetricByName(listMetricStore, ALL_TOTAL, 2);
+            testMetricByName(listMetricStore, RANGE_RATE, 2);
+            testMetricByName(listMetricStore, RANGE_TOTAL, 2);
+            testMetricByName(listMetricStore, FLUSH_RATE, 2);
+            testMetricByName(listMetricStore, FLUSH_TOTAL, 2);
+            testMetricByName(listMetricStore, RESTORE_RATE, 2);
+            testMetricByName(listMetricStore, RESTORE_TOTAL, 2);
+            return true;
+        } catch (final Throwable e) {
+            return false;
+        }
+    }
+
+    private boolean testCacheMetric() {
+        try {
+            final List<Metric> listMetricCache = new ArrayList<Metric>(kafkaStreams.metrics().values()).stream().filter(m -> m.metricName().group().equals(STREAM_CACHE_NODE_METRICS)).collect(Collectors.toList());
+            testMetricByName(listMetricCache, HIT_RATIO_AVG, 6);
+            testMetricByName(listMetricCache, HIT_RATIO_MIN, 6);
+            testMetricByName(listMetricCache, HIT_RATIO_MAX, 6);
+            return true;
+        } catch (final Throwable e) {
+            return false;
+        }
+    }
+
+    private boolean testMetricByName(final List<Metric> listMetric, final String metricName, final int numMetric) {
```

Comment:
The returned value of `testMetricByName` is not checked at all, so even if some metric name does not exist, the callers like `testCacheMetric` would still return true, right?

Reply:
My bad, you are right, i will review the tests

Reply:
Ah, i just fixed it, but not pushed